allScreensTogether.py

```diff
@@ -1,127 +1,108 @@
 
-        #instantiate the labels
-        title = Label(text="Select one of the data sets to load images",font_size=30, pos_hint={'x': 0, 'center_y': .9})
+        # instantiate the labels
+        title = Label(text="Select one of the data sets to load images", font_size=30,
+                      pos_hint={'x': 0, 'center_y': .9})
         cafeDescription = Label(text="*CAFE Description Here*", font_size=15, pos_hint={'x': -.2, 'center_y': .5})
         pofaDescription = Label(text="*POFA Description Here*", font_size=15, pos_hint={'x': .2, 'center_y': .5})
 
-        #add labels to the layout
+        # add labels to the layout
         layout.add_widget(title)
         layout.add_widget(cafeDescription)
         layout.add_widget(pofaDescription)
 
-        #binding the buttons to the clk method
+        # binding the buttons to the clk method
         cafeButton.bind(on_press=self.clk)
         pofaButton.bind(on_press=self.clk)
         loadButton.bind(on_press=self.changer)
 
-        return layout
+        self.add_widget(layout)
 
-    def changer(self,*args):
+    def changer(self, *args):
         self.manager.current = 'screen2'
 
-
     def clk(self, obj):
         print("Hello World!")
 
 
-    def callback():
-        print 'Button is being pressed.'
-        
-
-class TestApp(App):
-
-        def build(self):
-            my_screenmanager = ScreenManager()
-            screen1 = selectingDataSet(name='screen1')
-            screen2 = runPCA(name='screen2')
-            my_screenmanager.add_widget(screen1)
-            my_screenmanager.add_widget(screen2)
-            return my_screenmanager
-
-
-if __name__ == '__main__':
-    TestApp().run()
-
-
-
-
-
-
-
-
-
-class numComponentsSelect(FloatLayout):
-
+class NumComponentsSelect(FloatLayout):
     def __init__(self, **kwargs):
-        super(numComponentsSelect, self).__init__(**kwargs)
-        self.numComponents = Label(text='Number of components:  ',pos=(0,250))
-        #self.numComponents.bind(value=self.sliderChange)
-        compSlider = Slider(min=1, max=109,value=25,pos=(0,200))
+        super(NumComponentsSelect, self).__init__(**kwargs)
+        self.numComponents = Label(text='Number of components:  ', pos=(0, 250))
+        # self.numComponents.bind(value=self.sliderChange)
+        compSlider = Slider(min=1, max=109, value=25, pos=(0, 200))
         compSlider.bind(value=self.sliderChange)
         self.add_widget(self.numComponents)
         self.add_widget(compSlider)
 
         # after images
         layout = FloatLayout()
-        self.before1 = Image(source='face1.png',pos=(-300,100))
-        self.before2 = Image(source='face2.png',pos=(-170,100))
+        self.before1 = Image(source='face1.png', pos=(-300, 100))
+        self.before2 = Image(source='face2.png', pos=(-170, 100))
         layout.add_widget(self.before1)
         layout.add_widget(self.before2)
         self.add_widget(layout)
 
-    def sliderChange(instance,obj,value):
```

Comment:
Renamed `instance` to `self`, since that's usually how the self object is called...